Ok so there are 36 errors due to:

```
ValueError: underlying buffer has been detached
```

1 error due to:
```
_curses.error: must call (at least) setupterm() first
```

There's 1 FAIL:

```
FAIL: test_image_files (tests.test_renderers.TestRenderers)
Check that the ImageFile renderer works.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/asciimatics-1.10.0/tests/test_renderers.py", line 178, in test_image_files
    '     ..:::;Gr::s    '])
AssertionError: Lists differ: ['', '     sA3h3h3Hr2     ', '  ;:;G#99G@&2;;;r   ',[187 chars]   '] != ['', '        .:;rA       ', '    :2HG#;H2;s;;2   ',[187 chars]   ']

First differing element 1:
'     sA3h3h3Hr2     '
'        .:;rA       '

Diff is 691 characters long. Set self.maxDiff to None to see it.
```

It might be that the last 1 FAIL is actually expected to fail due to incompatible Pillow version. While the 36 errors are due to some weird STDOUT issue.

--- a/pkgs/development/python-modules/asciimatics/default.nix
+++ b/pkgs/development/python-modules/asciimatics/default.nix
@@ -1,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools_scm
+, pyfiglet
+, pillow
+, wcwidth
+, future
+, mock
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "asciimatics";
+  version = "1.10.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "9101b0b6885542f324980bbe13a772475cd6a12678f601228eaaea412db919ab";
+  };
+
+  nativeBuildInputs = [
+    setuptools_scm
+  ];
+
+  propagatedBuildInputs = [
+    pyfiglet
+    pillow
+    wcwidth
+    future
+  ];
+
+  checkInputs = [
+    mock
+    nose
+  ];
+
+  # tests require a pty emulator
+  # which is too complicated to setup here
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Helps to create full-screen text UIs (from interactive forms to ASCII animations) on any platform";
+    homepage = https://github.com/peterbrittain/asciimatics;
+    license = licenses.asl20;
+    maintainers = with maintainers; [ cmcdragonkai ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -176,6 +176,8 @@ in {
 
   asana = callPackage ../development/python-modules/asana { };
 
+  asciimatics = callPackage ../development/python-modules/asciimatics { };
+
   ase = callPackage ../development/python-modules/ase { };
 
   asn1crypto = callPackage ../development/python-modules/asn1crypto { };

